src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/TextBlock.cs

```diff
@@ -1868,7 +1868,7 @@ protected virtual IEnumerator<IInputElement> HostedElementsCore
                 if (_complexContent == null || !(_complexContent.TextContainer is TextContainer))
                 {
                     // Return empty collection
-                    return new HostedElements(new ReadOnlyCollection<TextSegment>(new List<TextSegment>(0)));
+                    return new HostedElements(ReadOnlyCollection<TextSegment>.Empty);
```

Comment:
@ThomasGoulet73 Could you revert the changes in Textblock? #9993 

Thank you.

Reply:
Sorry I missed your comment when answering below. Since it's the same changes it shouldn't conflict, I think it's fine for both PRs to stay as-is.

Reply:
I have asked because I personally try to not do overlapping changes (even if they would be the same) with other contributors open PRs when I'm aware of them existing (usually that includes search in PRs since I've got burnt on your PR many months ago, and this one would probably would go with "ReadOnlyCollection" search.). The other time you've asked me, now I'm politely asking you.

Not trying to be difficult, you may resolve the conversation as you see fit, it was just an ask since my changes were the driving subject of that PR and now it feels a bit wasted if it's already contained here.

Reply:
Sorry, I didn't notice that it was one of the main points of your PR. I thought the main point of your PR was using Span and the ReadOnlyCollection change was just a "While you're there" change. It wouldn't conflict because it's the same change but I'll revert it because I don't want to "erase" your work if my PR is merged first.